src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -888,9 +906,11 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
                         {
                             resolvesFeps = new List<FunctionEndPoint>() { compliantTarget };
                             replicationInstructions = replicationOption;
-                            return;
+                            matchFound = true;
                         }
                     }
+                    if (matchFound)
+                        return;
```

Comment:
@reddyashish so - what about case 5 - why do we exit the other cases late, and exit case 5 as soon as we find a match?

Reply:
I did not understand this case and it adds an empty replication option to the previous list and finds the match. I was not able to find any examples related to this case.

Reply:
I will try to find when it's called. 
@saintentropy any ideas?

Reply:
@mjkkirschner @reddyashish In general this whole function is still black magic to me.  I have been able to look at indivual examples and how they flow through but it is hard to make judgment on individual PR's without holistic picture.   I think we need to document examples of different data, data structures, replication settings, and functions and what the result replication instruction and function endpoint list.  Not sure if we need to do that on this PR but it is hard to validate the code changes otherwise

Reply:
For a start - I've filed a followup task to cover each of the missing cases with explicit tests.